Sources/SwiftSyntaxMacroExpansion/BasicMacroExpansionContext.swift

```diff
@@ -210,7 +210,9 @@ extension BasicMacroExpansionContext: MacroExpansionContext {
       // The syntax node came from the source file itself.
       rootSourceFile = directRootSourceFile
       offsetAdjustment = .zero
-    } else if let nodeInOriginalTree = sharedState.detachedNodes[Syntax(node)] {
+    } else if let nodeInOriginalTree = sharedState.detachedNodes[Syntax(node)]
+      ?? sharedState.detachedNodes[Syntax(node.root)]
+    {
```

Comment:
Good catch. I think this should just check `node.root` though, similar to `location(for:anchoredAt:fileName:)`

```suggestion
    } else if let nodeInOriginalTree = sharedState.detachedNodes[node.root]
    {
```